src/storage-client/src/source/persist_source.rs

```diff
@@ -340,6 +393,63 @@ fn downcast_stats<'a, T: Data>(stats: &'a dyn DynStats) -> Option<&'a T::Stats>
     }
 }
 
+impl PersistSourceDataStatsImpl<'_> {
+    fn col_json<'a>(&'a self, idx: usize, _arena: &'a RowArena) -> Option<ResultSpec<'a>> {
+        let name = self.desc.0.get_name(idx);
+        let stats = self
+            .stats
+            .key
+            .col::<Vec<u8>>(name.as_str())
+            .or_else(|_| {
+                self.stats
+                    .key
+                    .col::<Option<Vec<u8>>>(name.as_str())
+                    .map(|option| option.map(|s| &s.some))
+            })
+            .ok()??;
+        fn json_spec(stats: &JsonStats) -> ResultSpec {
+            match stats {
+                JsonStats::JsonNulls => ResultSpec::value(Datum::JsonNull),
+                JsonStats::Bools(bools) => {
+                    ResultSpec::value_between(bools.lower.into(), bools.upper.into())
+                }
+                JsonStats::Strings(strings) => ResultSpec::value_between(
+                    strings.lower.as_str().into(),
+                    strings.upper.as_str().into(),
+                ),
+                JsonStats::Numerics(floats) => {
+                    fn float_to_datum(float: f64) -> Datum<'static> {
+                        let numeric: Numeric = float.into();
+                        numeric.into()
+                    }
+                    ResultSpec::value_between(
+                        float_to_datum(floats.lower.floor()),
+                        float_to_datum(floats.upper.ceil()),
+                    )
+                }
+                JsonStats::Maps(maps) => {
+                    ResultSpec::map_spec(
+                        maps.into_iter()
+                            .map(|(k, v)| {
+                                // NB: we can't prove that a field is always present based on the stats
```

Comment:
is this still true? let's move it to a TODO(mfp) if it is and if we think there's some way to avoid this with more info in the stats

Reply:
Still true I think! Changed the comment.